### PR TITLE
feat: relax `Sized` bound on `Mutex`

### PR DESCRIPTION
**Pull request by Tropix126**
## Describe the changes this PR makes. Why should it be merged?
Makes `T` on `Mutex<T>` `?Sized`, relaxing the requirement that `T` have a known size at compile time. This matches the behavior of the standard library.

I think this is a breaking change? (although it shouldn't affect the vast majority of people)

### CODE REVIEW

File: packages/vexide-core/src/sync/mutex.rs [hunk 1]
```diff
@@ -83,12 +83,12 @@ impl<'a, T> Future for MutexLockFuture<'a, T> {
...
+pub struct Mutex<T: ?Sized> {
     raw: RawMutex,
     data: UnsafeCell<T>,
 }
-unsafe impl<T: Send> Send for Mutex<T> {}
-unsafe impl<T> Sync for Mutex<T> {}
+unsafe impl<T: ?Sized + Send> Send for Mutex<T> {}
+unsafe impl<T: ?Sized + Send> Sync for Mutex<T> {}
 
 impl<T> Mutex<T> {
```
[the remark this answers is not in the document]
Comment (Gavin-Niederman) [reply packages/vexide-core/src/sync/mutex.rs:93]: github kinda threw, the line im talking about is 93

[the remark this answers is not in the document]
Comment (Tropix126) [reply packages/vexide-core/src/sync/mutex.rs:93]: Mutexes can be coerced into an unsized type after creation. This is useful for trait objects - for example:
```rs
use std::sync::*;
use std::any::Any;

fn main() {
    let x: Arc<Mutex<dyn Any + 'static>> = Arc::new(Mutex::new(3));
}```

[the remark this answers is not in the document]
Comment (Gavin-Niederman) [reply packages/vexide-core/src/sync/mutex.rs:93]: :+1:

**Review by Gavin-Niederman — APPROVED** [review]
LGTM

**Pull request merged**